packages/composite-checkout/src/components/checkout-step.js

```diff
@@ -87,7 +87,7 @@ function CheckoutStepHeader( {
 			isActive={ isActive }
 			className={ joinClasses( [ className, 'checkout-step__header' ] ) }
 		>
-			<Stepper isComplete={ isComplete } isActive={ isActive }>
+			<Stepper isComplete={ isComplete } isActive={ isActive } stepNumber={ stepNumber || 0 }>
```

Comment:
If there are ever two steps which do not have step numbers, then this will be `0` for both of them, resulting in duplicate ids again.

It might be better to do something like change `CheckoutStepContainer` so that it passes its `id` prop through to `CheckoutStep` and use that to create the id of the `CheckIcon` instead.

Reply:
oh good call. I'll update it.

Reply:
Updated!